service/src/geopm_hash.cpp

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2015 - 2023, Intel Corporation
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+#include "config.h"
+#include "geopm_hash.h"
+
+#include <string.h>
+#include <zlib.h>
+#include <iostream>
+
+
+
+static uint64_t _crc32(uint64_t begin, uint64_t key)
+{
+    uint32_t key_32 = key;
```

Comment:
A bit surprised this assignment does *not* result in a data truncation warning/error: 64 bits to 32 bits.

Reply:
Do you think we need to cast it?  The compiler doesn't complain, and it seems clear to the reader to me.